source/directives/uiTreeNode.js

```diff
@@ -63,7 +63,7 @@
             });
 
             scope.$watch(attrs.expandOnHover, function(val) {
-              if ((typeof val) == 'boolean') {
+              if ((typeof val) == 'boolean' || (typeof val) == 'number') {
```

Comment:
@jacobscarter can you use `===` here? The return value of `typeof` is a string, so using the triple equals makes it clear that we're comparing strings here.